scripts/zkai_attention_kv_d16_quantized_softmax_receipt_gate.py

```
@@ -0,0 +1,610 @@
+
+from __future__ import annotations
+
+import argparse
+import copy
+import csv
+import json
+import pathlib
+import sys
+import tempfile
+from fractions import Fraction
+from typing import Any, Callable
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)